compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2216,9 +2233,14 @@ public boolean isTypeCastable(BLangExpression expr, BType sourceType, BType targ
             return true;
         }
 
+
+        if (sourceType.tag == TypeTags.ANY && targetType.tag == TypeTags.READONLY) {
+            return true;
+        }
         boolean validTypeCast = false;
 
-        if (sourceType.tag == TypeTags.UNION) {
+        // support anydata and json
```

Comment:
Shall we say something along the lines of 

```suggestion
        // Use instanceof to check for anydata and json.
```

Or it may not be clear why we are doing an instanceof check.